File: app/src/main/java/com/example/momobooklet_by_sm/manageuser/UserAccountsFragment.java
package com.example.momobooklet_by_sm.manageuser;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.momobooklet_by_sm.database.ViewModel.UserViewModel;
import com.example.momobooklet_by_sm.manageuser.UserProfileRecyclerViewAdapter;
import com.example.momobooklet_by_sm.database.model.UserModel;
import com.example.momobooklet_by_sm.databinding.FragmentUserAccountsBinding;

import java.util.ArrayList;
import java.util.List;


public class UserAccountsFragment extends Fragment {

UserViewModel mUserViewModel;
private FragmentUserAccountsBinding binding;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment

        UserModel object1= new UserModel("ABC MOBILE MONEY","76464","hello@gmail.com","pass1",false);
        UserModel object2= new UserModel("ABC MOBILE MONEY","76911464","hello@gmail.com","pass1",true);



        List<UserModel> list =new ArrayList<UserModel>();

list.add(object1);
list.add(object2);
        binding = binding.inflate(inflater, container, false);

        // Set up the RecyclerView
        RecyclerView recyclerView = binding.recyclerView;
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 1, GridLayoutManager.VERTICAL, false));
        UserProfileRecyclerViewAdapter adapter = new UserProfileRecyclerViewAdapter(list);
        recyclerView.setAdapter(adapter);
        // add dummy data in database

        mUserViewModel = new ViewModelProvider(this ).get(UserViewModel.class);
        AddUsers();
        return binding.getRoot();
    }


    private void  AddUsers(){

        UserModel object1= new UserModel("ABC MOBILE MONEY","76464","hello@gmail.com","pass1",false);
        UserModel object2= new UserModel("ABC MOBILE MONEY","76911464","hello@gmail.com","pass1",true);


mUserViewModel.addUser(object1 );
mUserViewModel.addUser(object2);
        Toast.makeText(getContext(),"users added",Toast.LENGTH_SHORT).show();

    }
}

File: app/src/main/java/com/example/momobooklet_by_sm/manageuser/UserAccountsFragment.kt
package com.example.momobooklet_by_sm.manageuser
import com.example.momobooklet_by_sm.ui.viewmodels.UserViewModel
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.lifecycle.ViewModelProvider
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import com.example.momobooklet_by_sm.R
import com.example.momobooklet_by_sm.databinding.FragmentUserAccountsBinding


class UserAccountsFragment : Fragment() {

    private val mUserViewModel: UserViewModel by lazy{
        ViewModelProvider(this)[UserViewModel::class.java]
    }
    private lateinit var _binding: FragmentUserAccountsBinding
    private val binding get() = _binding
    private lateinit var adapter : UserProfileRecyclerViewAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        _binding = FragmentUserAccountsBinding.inflate(inflater, container, false)
        adapter = UserProfileRecyclerViewAdapter(mUserViewModel)

        // Set up the RecyclerView
        val recyclerView = binding.recyclerView
        recyclerView.adapter = adapter

        ShowTransactions()
        setAddAccountListener()
        return binding.root
    }
    private fun setAddAccountListener() {
        binding.addAccBtn.setOnClickListener {
            it!!.findNavController()
                .navigate(R.id.action_userAccountsFragment_to_registerFragment)
        }
    }
    /*populates recyclerview*/
    private fun ShowTransactions() {
        mUserViewModel.readAllData.observe(viewLifecycleOwner) { users ->
            adapter.differ.submitList(users)
        }
    }
}